cli/injector/main.go

```diff
@@ -317,11 +317,7 @@ func initPodWatch(resourceVersion string) (<-chan watch.Event, error) {
 func inject(kind string, sendToMetrics bool, reinjection bool) bool {
 	errOnInject := false
 
-	for i, inj := range injectors {
-		if onInit && len(configs) > i && configs[i].TargetContainer != nil && configs[i].TargetContainer.Name() == chaosInitContName {
```

Comment:
The faulty condition.

When using on Init we have only the chaos-handler init container as a targeted container at the beginning, because the other containers of the pod are not ready yet thus not detected.